### ISSUE TITLE
Able to undo past initial state

### ISSUE THREAD

**Issue by ehshi1992** [automerge/automerge#219]
I am trying to utilize the libraries undo functionality, however, `Automerge.canUndo` is returning true when called immediately after the initialization of my document.
```
    this.actionDoc = Automerge.from({
      clips: initialClips,
    })
    Automerge.canUndo(this.actionDoc)
```

Displaying the history of the document immediately after initialization only shows the "Initialization" action. Yet, canUndo still returns true.

My question is why it makes sense to be able to undo the populating of the initial state? Is there anyway to change the undo threshold?

**Comment by ept** [comment]
Hi @ehshi1992, good point. This is not deliberate, just an artifact of how initialization is implemented. If you think we should change this, we can. Do any others have opinions?

To implement this, I think would need to add an option (e.g. `undoable: false`) to the change requests sent from frontend to backend, and the backend already has a facility for applying a change without creating an undo history entry.

**Comment by ehshi1992** [comment]
Adding an optional flag sounds great. For my use case, I only need to prevent the initial state being undone, but it sounds like you guys can knock this out as a feature along with the fix!

**Comment by ept** [comment]
I have put up a proposed patch at #222.

**Comment by ept** [comment]
Merged #222, closing this issue.
